src/main.rs

```diff
@@ -47,13 +48,11 @@ fn main() {
     if matches.opt_present("h") {
         return println!("{}", usage(opts));
     }
-
-    let mut out: Box<Write> = match matches.free.first() {
-        None => Box::new(io::stdout()),
-        Some(file) => {
-            let append = matches.opt_present("a");
-            Box::new(open_file(file, append))
-        }
-    };
-    pipe(&mut io::stdin(), &mut out);
+    let append = matches.opt_present("a");
+    if matches.free.len() > 1 {
+        eprintln!("{}", usage(opts));
+        process::exit(1);
+    }
+    let path = matches.free.first().map(Borrow::borrow);
```

Comment:
Shouldn't we unwrap this?

Reply:
It's what I did initially, but doing so the user needs to specify a file, instead of having the output written on the `stdout`. So I changed it [here](https://github.com/jridgewell/rw/pull/2/commits/afdd1d1af5628571b63d475b474cea3b37107b05) to bring the feature back.

Now the `path` is an `Option<&str>` and the `pipe` function writes to `stdout` if the file param is not specified.

(Note: the `.map(Borrow::borrow)` is there to convert from `Option<&String>`, which is the output of `matches.free.first()`, into `Option<&str>`, which seems to me a more idiomatic type to use as `pipe` parameter. `.map(|it| it as &str)` would have worked as well)